### PR TITLE
[CI:BUILD] Packit: add Fedora downstream tasks

### PR DESCRIPTION
**Pull request by lsm5**
Get rid of `netavark.spec.rpkg` in favour of
`rpm/netavark.spec` which gets synced with fedora dist-git on every upstream release. The version in the new spec file is set to `0` by default and gets updated by packit automatically on every packit task.

Packit will automatically create a PR on fedora dist-git on every new upstream release. A sample PR will look like:
https://src.fedoraproject.org/rpms/container-selinux/pull-request/10#

A dry run for this can be triggered using:
`$ packit propose-downstream --local-content`

To run this command locally, you would need to have your packit user-configuration-file set.
Ref: https://packit.dev/docs/configuration/#user-configuration-file

along with a fedora api key created at:
https://src.fedoraproject.org/settings#nav-api-tab with sufficient ACLs.

With this PR, downstream Fedora spec will get synced with `rpm/netavark.spec` present upstream, so upstream will remain independent of downstream and the packaging process will become a part of upstream maintenance. The downstream tasks add zero extra cost to upstream release process (except an automatic github issue on downstream task failure).

### CODE REVIEW

**Comment by lsm5** [comment]
Copr builds look good: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/packit-builds/build/6090069/
koji scratch build also looks good: https://koji.fedoraproject.org/koji/taskinfo?taskID=102222423

Ready to go.

@Luap99 @mheon @baude @flouthoc PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

File: rpm/netavark.spec [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+tar fx %{SOURCE1}
+mkdir -p .cargo
+
+cat >.cargo/config << EOF
+[source.crates-io]
+replace-with = "vendored-sources"
+
+[source."https://github.com/containers/netavark-dhcp-proxy"]
+git = "https://github.com/containers/netavark-dhcp-proxy"
+replace-with = "vendored-sources"
```
Comment (Luap99) [inline rpm/netavark.spec:102]: this can be removed now, the repo is no longer used and was merged directly into nv

Comment (lsm5) [reply rpm/netavark.spec:102]: ohh nice. will update PR in a bit, thanks

Comment (lsm5) [reply rpm/netavark.spec]: done. PTAL.

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**